src/match.c

```diff
@@ -1,128 +1,199 @@
-{
-    matches = g_list_append(matches, newMatch);
-    return matches;
-}
-
-//changed inputs, return type
-GList* remove_at(int index, GList* matches)
-{
-    //get element at position
-    GList *toDelete = g_list_nth(matches, index);
-    
-    //remove position at element
-    matches = g_list_delete_link(matches, toDelete);
-    return matches;
-}
-
-//changed inputs
-match* get_at_index(int index, GList* matches)
-{
-    if (g_list_length(matches) == 0) return NULL;
-    GList *toReturn = g_list_nth(matches, index);
-    if (toReturn == NULL) return NULL;
-    return toReturn -> data;
-}
-
-//changed inputs
-int get_index(match* match, GList* matches)
-{
-    int position = g_list_index(matches, match);
+int match_get_line_num(match* match) {
+    if (match == NULL) {
+        return -1;
+    }
+    return match->lineNum;
+}
+
+int match_get_position(match* match) {
+    if (match == NULL) {
+        return -1;
+    }
+    return match->position;
+}
+
+
+match* match_next(int index, list_t* matches) {
+    int size = list_size(matches);
+    return list_get_at(matches, (index + 1) % size);
+}
+
+
+match* match_prev(int index, list_t* matches) {
+    int size = list_size(matches);
+    return list_get_at(matches, ((index - 1) + size) % size);
+}
+
+
+void match_insert_at(match* newMatch, int index, list_t* matches) {
+    if (index > (int) (list_size(matches) - 1)) {
+        list_append(matches, newMatch);
+    }
+    if (index < 0) {
+        list_prepend(matches, newMatch);
+    }
+    list_insert_at(matches, newMatch, index);
+}
+
+
+void match_append(match* newMatch, list_t* matches) {
+    list_append(matches, newMatch);
+}
+
+
+match* match_get_at_index(int index, list_t* matches) {
+    if (list_size(matches) == 0) {
+        return NULL;
+    }
+    match* cur = list_get_at(matches, index);
+    return cur;
+}
+
+
+int match_get_index(match* match, list_t* matches) {
+    int position = list_locate(matches, match);
     return position;
 }
 
-//just for testing
-void pretty_print(GList* l)
-{
-    GList *list = l;
-    while (list != NULL){
-        GList *next = list->next;
-        printf("list data: %s\n", (char*)list->data);
-        list = next;
+void list_info(list_t* l) {
+    printf("\n");
+    printf("... displaying info about list ...\n");
+    printf("The list now holds %u elements:\n", list_size(l));
+
+    list_iterator_start(l);               /* starting an iteration "session" */
+
+    while (list_iterator_hasnext(l)) {   // tell whether more values available
+        match cur = *(match *)list_iterator_next(l); /* get the next value */
+        match_display(&cur);
     }
+
+    list_iterator_stop(l);                 /* ending the iteration "session" */
+
+    printf("... finished ...\n");
+}
+
+
+void ftsh_set_color_red() {
+    printf("\033[1;31m");
+}
+
+
+void ftsh_reset_color() {
+    printf("\033[0m");
+}
+
+
+void match_print_line(match* match) {
+    int i = 0;
+    int wl = 0;
```

Comment:
Let's have some more descriptive variable names in this function.

Reply:
done!